### DISCUSSION TITLE
Support for QUIC transport protocol?

### DISCUSSION THREAD

**Discussion by sarsonj** [processone/ejabberd#3580] [Ideas]
Hi,

ejabberd currently works over standard TCP socket, Websockets and Bosh. However, do you guys considering to adapt QUIC protocol? It should help to make much faster reconnection, because of reduced number of roundtrips for establishing secure connection.

Jindrich

**Comment by mremond** [comment]
Would that really make a big difference, given that XMPP is a connected protocol, based on long-running sessions ?

**Reply by sarsonj** [reply]
It makes big difference on mobile - where is frequent connecting / disconnecting (usually app disconnect when going to background and connect when back in foreground). It would help to shrink connection times - now it is few seconds with standard login.

Also QUIC don't terminate connection when device switches network - like from WiFi to LTE or between 2 WiFis.
